state/event_table.go

```diff
@@ -259,6 +259,34 @@ func (t *EventTable) SelectStrippedEventsByIDs(txn *sqlx.Tx, verifyAll bool, ids
 
 }
 
+// SelectUnknownEventIDs accepts a list of event IDs and returns the subset of those which are not known to the DB.
+func (t *EventTable) SelectUnknownEventIDs(txn *sqlx.Tx, maybeUnknownEventIDs []string) (map[string]struct{}, error) {
+	// Note: in practice, the order of rows returned matches the order of rows of
+	// array entries. But I don't think that's guaranteed. Return an (unordered) set
+	// out of paranoia.
+	queryStr := `
+	WITH maybe_unknown_events(event_id) AS (SELECT unnest($1::text[]))
+	SELECT event_id
+	FROM maybe_unknown_events LEFT JOIN syncv3_events USING(event_id)
+	WHERE event_nid IS NULL;`
+
+	var unknownEventIDs []string
+	var err error
+	if txn != nil {
```

Comment:
Why allow this to be nil? We call it with a txn.

Reply:
Was cribbing from https://github.com/matrix-org/sliding-sync/blob/f28f7d0599a91d9f424eaee4e9b4c3c9f9ae9550/state/event_table.go#L188-L202

and erring towards paranoia. Can remove.